test/scenario_runner_test.exs

```diff
@@ -105,6 +116,34 @@ defmodule WhiteBread.Runners.ScenarioRunnerTest do
     assert result == :ok
   end
 
+  test "Contexts can run finalization provided by scenario_finalize method for both failed and normal steps" do
```

Comment:
Can this be split in to two separate tests? I think I'd rather have them fail individually.